### ISSUE TITLE
Clarify Indexing

### ISSUE THREAD

**Issue by kianenigma** [paritytech/contract-docs#28]
I am aware that an indexer support is WIP, but noting it here to clarify in the future. 

Once I deploy a contract, my intuitive next step is to go to subscan and see some evidence of what I have done. 

More confusing, remix will print a block number and hash at which this contract is deployed, which I don't really know where it is coming from. 

Would be good to have a banner in this tutorial, clarifying the indexing situation.

**Comment by kianenigma** [comment]
Can someone in the meantime tell me what is the overall expectation here? Do we expect users to use any of the substrate based explorers, or ETH ones? 

Another issue I have noticed is that StateScan and Subscan have very different data. 

<img width="2235" alt="Screenshot 2024-12-11 at 11 52 44" src="https://github.com/user-attachments/assets/77eb01ce-cc82-469b-a16b-bd3ba6ff48f3">

About naming in StateScan: https://github.com/opensquare-network/statescan-v2/issues/984 they said they will fix it asap. 

We can likely ask one to be offline to avoid confusion.

**Comment by athei** [comment]
statescan seems to not show unsigned transactions. The `eth_transact` call is unsigned and is used to submit eth transactions to chain. The other calls are done by normal signed ethereum transactions. Basically when you are interacting with `pallet_revive` using an Polkadot wallet. Those txes were made by me using pjs apps.

**Comment by wliyongfeng** [comment]
We implemented EVM scan recently for laos network. Check an example tx [here](https://laos.statescan.io/#/tx/0x3aecdcf465403f2dd921f01650123d8702f9a930c0a4dc8f83ab5f466309e672). Has westend assethub already supported EVM?

**Comment by wliyongfeng** [comment]
> I assume you are decoding the block contents manually and then decoding the eth payload in there

No. We are calling data from a EVM RPC, code [here](https://github.com/opensquare-network/statescan-v2/blob/main/backend/packages/block-scan/src/scan/events/evm/tx/index.js#L50), and combine it with an extrinsic indexer.


> I believe a better way is to do it exclusively via the `debug_` tracing RPC endpoints we will be adding

Do you mean a EVM compatible RPC for this tracing RPC?

**Comment by pgherveou** [comment]
> Do you mean a EVM compatible RPC for this tracing RPC?

Yes that's the plan, although I don't think there are official specs for tracing endpoints so we will follow what geth is doing

**Comment by wliyongfeng** [comment]
> Yes that's the plan, although I don't think there are official specs for tracing endpoints so we will follow what geth is doing

Let us know updates and feel free to leave issues on [statescan](https://github.com/opensquare-network/statescan-v2/issues), so we can support ASAP.
